eng/pipelines/pr-breaking-check.yml

```diff
@@ -0,0 +1,94 @@
+              $packageJson.devDependencies["@typespec/compiler"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/compiler"
+              $packageJson.devDependencies["@typespec/http"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/http"
+              $packageJson.devDependencies["@typespec/openapi"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/openapi"
+              $packageJson.devDependencies["@typespec/openapi3"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/openapi3"
+              $packageJson.devDependencies["@typespec/rest"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/rest"
+              $packageJson.devDependencies["@typespec/versioning"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/versioning"
+
+              $packageJson | ConvertTo-Json -Depth 100 | Out-File -Path "package.json" -Encoding utf8 -NoNewline -Force
+            displayName: "Update package.json in azure-rest-api-specs"
+            workingDirectory: $(Build.SourcesDirectory)/azure-rest-api-specs
+          - script: npm install
+            displayName: Install updated TypeSpec dependencies
+            workingDirectory: $(Build.SourcesDirectory)/azure-rest-api-specs
+          - pwsh: ./eng/scripts/TypeSpec-Generate-Swagger.ps1 -CheckAll
+            displayName: "Generate all swagger files"
+            workingDirectory: $(Build.SourcesDirectory)/azure-rest-api-specs
+          - pwsh: |
+              git restore package.json
+              git restore package-lock.json
+            displayName: Discard versions change
+            workingDirectory: $(Build.SourcesDirectory)/azure-rest-api-specs
+
+          # Step 3: Create PR
+          - checkout: azure-sdk-tools
+            fetchDepth: 1
+          - template: /eng/common/pipelines/templates/steps/git-push-changes.yml@azure-sdk-tools
+            parameters:
+              BaseRepoBranch: auto-update-swagger-$(Build.BuildNumber)
+              BaseRepoOwner: azure-sdk
+              CommitMsg: Update swagger with TypeSpec built from $(Build.BuildNumber)
+              TargetRepoOwner: Azure
+              TargetRepoName: azure-rest-api-specs
+              WorkingDirectory: $(Build.SourcesDirectory)/azure-rest-api-specs
+              ScriptDirectory: $(Build.SourcesDirectory)/azure-sdk-tools/eng/common/scripts
+          - task: PowerShell@2
+            displayName: Create pull request
+            inputs:
+              pwsh: true
+              filePath: $(Build.SourcesDirectory)/azure-sdk-tools/eng/common/scripts/Submit-PullRequest.ps1
+              arguments: >
+                -RepoOwner "Azure"
+                -RepoName "azure-rest-api-specs"
+                -BaseBranch "main"
+                -PROwner "azure-sdk"
+                -PRBranch "auto-update-swagger-$(Build.BuildNumber)"
```

Comment:
This naming pattern for a branch will create a branch/PR per build in the specs repo which is something we should definitely avoid, that will likely cause way too much noise.

Reply:
In the typespec preview pipelines, we set the branch name early in the build via inline powershell.   We attempt to reuse the same PR branch name for the same source PR, scheduled runs use a single branch name and all other runs will use a unique dynamic name:

[azure-sdk-tools / archetype-autorest-preview.yml](https://github.com/Azure/azure-sdk-tools/blob/main/eng/pipelines/templates/stages/archetype-autorest-preview.yml)

```yaml
    - pwsh: |
        $sourceBranch = '$(Build.SourceBranch)'
        $buildReason = '$(Build.Reason)'
        $buildNumber = '$(Build.BuildNumber)'

        if ($buildReason -eq 'Schedule') {
          $branchName = 'auto-update-autorest-scheduled'
        } elseif ($sourceBranch -match "^refs/pull/(\d+)/(head|merge)$") {
          $branchName = "auto-update-autorest-pr-$($Matches[1])"
        } else {
          $branchName = "auto-update-autorest-$buildNumber"
        }

        Write-Host "Setting variable 'branchName' to '$branchName'"
        Write-Host "##vso[task.setvariable variable=branchName;isOutput=true]$branchName"
      displayName: Set branch name
      name: set_branch_name
```

Reply:
That would definitely be better if we need a PR. However I'd still like to understand if a PR into the specs repo is needed to accomplish the testing we are looking for.

Reply:
We want to have a visualized way to see the diff. We could reduce the number by the way Patrick put above. Or maybe we could even put the PRs in azure-rest-api-specs-pr if your concern is you don't want to confuse customers.

> If it is needed who will be responsible for monitoring it and closing it.

@hallipr do you know how it is monitored in azure-sdk-for-net?

Reply:
You can have a visual change by pushing to a branch without needing to create a PR. Creating a PR causes a lot of other things to kick off that we should avoid if they aren't needed.

Noise inside of the private repo is still a concern.

Reply:
How about we just create a closed PR? It sounds really wired to me that pushing to a branch which is almost 100% going to create a PR.

Reply:
@pshao25 we can compose the URL to view the difference between two branches like this `https://github.com/[username]/[repository]/compare/[base-branch]...[compare-branch]` instead of creating PRs.

Reply:
If the purpose of the branch (or pr) is to use github's UI to visualize a diff, I like Ray's suggestion to just use a ref comparison instead of a PR.   We could even add that compare URL to the pipeline output so it's clickable from the ADO UI